### ISSUE TITLE
really long (infinite?) processing of large highly cyclic smile

### ISSUE THREAD

**Issue by dkoes** [rdkit/rdkit#1360]
When trying to Chem.MolFromSmiles this beautiful molecule, rdkit took 2+ hours and used up 200+ GB of memory before I killed it.
`COC1=CC=C(C=C1)C2C3=C(C=CC4=CC=CC=C43)OC5=CC6=C(C=C5)C7=NC8=C9C=CC1=CC9=C(N8)N=C3C4=C5C=CC(=C4)OC4=C(C(C8=C(C=CC9=CC=CC=C98)OC8=CC9=C(C=C8)C8=NC9=NC9=C%10C=C(C=CC%10=C(N9)N=C9C%10=C(C=C(C=C%10)OC%10=C2C2=CC=CC=C2C=C%10)C(=N9)NC2=NC(=N8)C8=C2C=C(C=C8)OC2=C(C(C8=C(C=CC9=CC=CC=C98)OC8=CC9=C(C=C8)C(=NC5=N3)N=C9NC6=N7)C3=CC=C(C=C3)OC)C3=CC=CC=C3C=C2)OC2=C(C(C3=C(O1)C=CC1=CC=CC=C13)C1=CC=C(C=C1)OC)C1=CC=CC=C1C=C2)C1=CC=C(C=C1)OC)C1=CC=CC=C1C=C4	PubChem-102128817`

I observed this with the Ubuntu 16.04 repository version (2015.03.1) and verified the problem still exists in upstream/master (2017.03.1.dev1).

**Comment by greglandrum** [comment]
Lovely.
That's the ring finding code. 
I *really* hope to have this resolved by the next release, but it may end up not making it.

**Comment by greglandrum** [comment]
In the meantime, if you actually want to do something with that molecule, let me know: there is a way to process it.

**Comment by dkoes** [comment]
Nope, I pass over the molecule as soon as I realize it's too big. I now check the string length before trying to parse to avoid this issue.

**Comment by tentrillion** [comment]
Seems like this is an exact duplicate of https://github.com/rdkit/rdkit/issues/1281 .  I guess this crazy molecule is getting popular!

**Comment by dkoes** [comment]
FYI, this smile now fails to parse with this error message, which is probably acceptable:
```
[09:47:28] Maximum BFS search size exceeded.
This is likely due to a highly symmetric fused ring system.
```

**Comment by greglandrum** [comment]
> FYI, this smile now fails to parse with this error message, which is probably acceptable:
> 
> ```
> [09:47:28] Maximum BFS search size exceeded.
> This is likely due to a highly symmetric fused ring system.
> ```

Thanks for that update @dkoes

**Comment by oiao** [comment]
Just stumbled upon this as well. Is there perhaps a better way to handle this? Perhaps throwing a dedicated exception?

Edit: Happy to submit a PR, if a dedicated exception makes sense here. Or would you just suggest to stick to more explicit handling downstream, such as
```python
>>> mol = Chem.MolFromSmiles(sm, sanitize=False)
>>> error = Chem.SanitizeMol(mol, catchErrors=True)
>>> error
rdkit.Chem.rdmolops.SanitizeFlags.SANITIZE_SYMMRINGS
```
